### DISCUSSION TITLE
Add additional Filament type

### DISCUSSION THREAD

**Discussion by Billbo911** [Jyers/Marlin#1712] [Q&A]
I have a simple question, or maybe it's a feature request. That will depend on the answer to my question.

I am currently running E3V2-BLTouch-5x5-HS-v4.2.2-v2.0.1.bin. I see in the menu for both Pre-heat and Temperature, there are only 3 Filament types listed: PLA, PETG, and ABS. That is nice, but can additional types be added, say TPU?

If this capability is already there, I apologies. Please direct me to where I can find the info on how to add the additional types. Otherwise, can this capability be added?

**Comment by Cal-LF** [comment]
The TPU filament is in the config.h file but not defined, so as of current release, you have to compile your own to activate the 5th filament preheat to the menu.

![filament](https://user-images.githubusercontent.com/76499230/147171217-08135884-268f-497e-ae69-be6892a5b4ce.jpg)
.
